app/controllers/concerns/authenticate.rb

```diff
@@ -27,4 +33,24 @@ def user_signed_in?
   def authenticate_user!
     redirect_to login_path unless current_user
   end
+
+  private
+
+  def find_current_user_based_on_session
+    Current.user = User.find_by(id: session[:current_user_id])
+    Current.person = Current.user&.person
+    Current.user = nil if Current.person.nil?
+  end
+
+  def find_current_user_based_on_http_header
+    if Feature.registration?
+      Current.user = User.find_or_create_by!(auth_uid: request.headers[Setting.http_header_auth_uid]) do |user|
+        user.create_person!(email: request.headers[Setting.http_header_auth_email])
+        user.name = request.headers[Setting.authentication_http_header_name] || person.email
+      end
+    else
+      Current.user = User.find_by(auth_uid: request.headers[Setting.http_header_auth_uid])
+    end
+    Current.person = Current.user&.person
+  end
```

Comment:
TODO: This private helper method should probably end with the same consistency check `Current.user = nil if Current.person.nil?`